packages/sdk/ts/user/utils.ts

```diff
@@ -243,6 +254,12 @@ export const getPollJoiningCircuitEvents = async ({
     sleepAmount: 0,
   });
 
+  const loadedStateIndex = getStateIndex(signUpData.pubKeys, userMaciPubKey, stateIndex);
```

Comment:
why do we need userMaciPubKey? we already have the private and we can derive the public as in line 263